Doc/library/os.rst

```diff
@@ -3780,6 +3780,174 @@ features:
    .. versionadded:: 3.10
 
 
+.. function:: timerfd_create(clockid, flags)
+
+   Create and return a timer file descriptor. The file descriptors supports
+   raw :func:`read` with a buffer size of 8, :func:`~select.select`, :func:`~select.poll`
+   and similar. See man page :manpage:`timerfd_create(2)` for more information.
+
+   Refer to :ref:`time-clock-id-constants` for a list of accepted values for *clk_id*.
+
+   *flags* can be constructed from :const:`TFD_NONBLOCK` and :const:`TFD_CLOEXEC`.
+
+   If the timerfd counter is zero and :const:`TFD_NONBLOCK` is not
+   specified, :func:`read` blocks.
+
+   Example::
+
+       import time
+       import os
+
+       fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+
+       # timer interval 0.5 second
+       inteval = 0.5
+
+       # timer start in 1 second
+       value = 1
+
+       # start timer
+       _, _ = os.timerfd_settime(fd, 0, inteval, value)
+
+       try:
+           while:
+               # wait timerfd expired
+               _ = fd.read(8)
+               do_work()
+       finally:
+           os.close(fd)
+
+   Example::
+
+       import time
+       import os
+
+       fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+
+       # timer interval 0.5 second
+       inteval_ns = 10**9 // 2
+
+       # timer start in 1 second
+       value_ns = 10**9
+
+       # start timer in nano second unit
+       _, _ = os.timerfd_settime_ns(fd, 0, inteval_ns, value_ns)
+
+       try:
+           while:
+               # wait timerfd expired
+               _ = fd.read(8)
+               do_work()
+       finally:
+           os.close(fd)
+
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. function:: timerfd_settime(fd, flags, interval, value)
+
+   Start/stop timer for an timerfd file descriptor.
+   See man page :manpage:`timerfd_settime(2)` for more information.
+
+   ``interval`` coresponds to ``it_interval`` in ``struct itimerspec`` and ``value`` coresponds to
+   ``it_value`` in ``struct itimerspec``. They are both in seconds unit and types are double.
+
+   ``interval`` is calculated like ``it_interval.tv_sec + it_interval.tv_nsec * 1e-9``,
+   ``value`` is calculated like ``it_interval.tv_sec + it_interval.tv_nsec * 1e-9`` internally.
```

Comment:
The params should be marked up as `*param_name*`, the C struct members with ```:c:member:`!member` ```, and the C structs with ```:c:struct:`!struct s` ```.

I guess we could use `c:expr:` for the calculations, but I'm not sure about the Sphinx magic that happens with `c:expr:`. @AA-Turner would know.

Please also document the return value.

Reply:
`:c:expr:` adds cross references to everything it can and leaves the rest with literal formatting (similar to using ``` `` ```). `c:texpr:` can be used to render it as normal text instead.

Reference: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#inline-expressions-and-types

A

Reply:
@AA-Turner @erlend-aasland 
Fixed.